sw/device/lib/pinmux.c

```diff
@@ -41,23 +62,16 @@ void pinmux_init(void) {
                  PINMUX_MIO_OUTSEL_0_OUT_0_MASK,
                  PINMUX_PERIPH_OUTSEL_IDX_OFFSET);
 
-  // Configure UART RX input to connect to MIO pad IOR2
-  mmio_region_t reg32 = mmio_region_from_addr(
-      PINMUX0_BASE_ADDR + PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET);
-  uint32_t reg_value = kTopEarlgreyPinmuxInselIoc10;
-  // We've got one insel configuration field per register. Hence, we have to
-  // convert the enumeration index into a byte address using << 2.
-  uint32_t reg_offset = kTopEarlgreyPinmuxPeripheralInUart0Rx << 2;
-  uint32_t mask = PINMUX_MIO_PERIPH_INSEL_0_IN_0_MASK;
-  mmio_region_write32(reg32, reg_offset, reg_value & mask);
-
-  // Configure UART TX output to connect to MIO pad IOR3
-  reg32 =
-      mmio_region_from_addr(PINMUX0_BASE_ADDR + PINMUX_MIO_OUTSEL_0_REG_OFFSET);
-  reg_value = kTopEarlgreyPinmuxOutselUart0Tx;
-  // We've got one insel configuration field per register. Hence, we have to
-  // convert the enumeration index into a byte address using << 2.
-  reg_offset = kTopEarlgreyPinmuxMioOutIoc11 << 2;
-  mask = PINMUX_MIO_OUTSEL_0_OUT_0_MASK;
-  mmio_region_write32(reg32, reg_offset, reg_value & mask);
+  init_uart_pins(
```

Comment:
Should we initialize these ONLY in the UART test? pinmux_init is called by other tests that have nothing to do with the UARTs.

Reply:
yeah we're going to discuss this on monday. I think we need to stop overloading these common functions and split them out to common test routines.

Reply:
I saw Pirmin connected the uart1 to another pins. I end up moving all these pinmux initialization for uart to the uart test itself. Once we will come up with a pinmux configuration, I will update accordingly.